cli/unipipe/commands/git.ts

```diff
@@ -0,0 +1,99 @@
+import { Command } from "../deps.ts";
+import { Repository } from "../repository.ts";
+
+interface GitOpts {
+  name: string;
+  email: string;
+  message: string;
+}
+
+export function registerGitCmd(program: Command) {
+  program
+    .command("git <cmd> [repo]")
+    .description("Runs Git to simplify pull/push commands")
+    .action(async (opts: GitOpts, cmd: string, repo: string | undefined) => {
+
+      const repository = new Repository(repo ? repo : ".");
+      opts.name = opts.name || "Unipipe CLI";
+      opts.email = opts.email || "unipipe-cli@meshcloud.io";
+      opts.message = opts.message || "Commit changes";
+
+      switch (cmd) {
+        case "pull":
+          await commandPull(repository);
+          break;
+        case "push":
+          await commandPush(repository, opts);
+          break;
+        default:
+          console.log(`Git command '${cmd}' is not found`);
+      }
+    });
+}
+
+async function commandPull(repo: Repository) {
+  const pullFastForward = await gitPullFastForward(repo.path);
+
+  if (!pullFastForward) {
+    await gitPullRebase(repo.path);
```

Comment:
d: does the rebase work successfully if there are local changes? I guess there should only be local changes on files that haven't been updated remotely. So there should be ne real git conflict. Does i work successfully in that case?

Reply:
No, it will not work. To allow this we can stash the changes before pull and restore them afterward.